src/bp/nlu-core/training-pipeline.ts

```diff
@@ -510,11 +510,19 @@ const TrainOutOfScope = async (input: TrainStep, tools: Tools, progress: progres
 
 const NB_STEPS = 6 // change this if the training pipeline changes
 
-export const Trainer = async (
-  input: TrainInput,
-  tools: Tools,
-  progress?: (x: number) => void
-): Promise<TrainOutput> => {
+type Func = (...args: any[]) => any
+const makeLogDecorator = (trainId: string, logger?: sdk.NLU.Logger) => {
```

Comment:
I'm kindof against calling this a decorator if you don't use it as a decorator.

I know decorators are simply wrapper functions but most of the time they don't hate decorator in their naming, that would imply that it can strictly be used as decorator

Reply:
I would use it as a decorator, but TypeScript doesn't allow it.

Do you want me to just remove the word "Decorator" from the function name? Do you want me to use `makeLogger` or `makeLogWrapper` ?